roles/basic_security/tasks/ssh_and_sudo.yml

```diff
@@ -0,0 +1,85 @@
+    name: "{{ basic_security_sshd_name }}"
+    state: "{{ basic_security_sshd_state }}"
+  become: true
+
+- name: Update SSH configuration to be more secure.
+  ansible.builtin.lineinfile:
+    dest: "{{ basic_security_ssh_config_path }}"
+    regexp: "{{ item.regexp }}"
+    line: "{{ item.line }}"
+    state: present
+    validate: 'sshd -T -f %s'
+    mode: '0644'
+  with_items:
+    - regexp: "^PasswordAuthentication"
+      line: "PasswordAuthentication {{ basic_security_ssh_password_authentication }}"
+    - regexp: "^PermitRootLogin"
+      line: "PermitRootLogin {{ basic_security_ssh_permit_root_login }}"
+    - regexp: "^Port"
+      line: "Port {{ basic_security_ssh_port }}"
+    - regexp: "^UseDNS"
+      line: "UseDNS {{ basic_security_ssh_usedns }}"
+    - regexp: "^PermitEmptyPasswords"
+      line: "PermitEmptyPasswords {{ basic_security_ssh_permit_empty_password }}"
+    - regexp: "^ChallengeResponseAuthentication"
+      line: "ChallengeResponseAuthentication {{ basic_security_ssh_challenge_response_auth }}"
+    - regexp: "^GSSAPIAuthentication"
+      line: "GSSAPIAuthentication {{ basic_security_ssh_gss_api_authentication }}"
+    - regexp: "^X11Forwarding"
+      line: "X11Forwarding {{ basic_security_ssh_x11_forwarding }}"
+  become: true
+  notify: basic_security_restart_sshd
+
+- name: Add configured users allowed to connect over ssh.
+  ansible.builtin.lineinfile:
+    dest: "{{ basic_security_ssh_config_path }}"
+    regexp: '^AllowUsers'
+    line: "AllowUsers {{ basic_security_ssh_allowed_users | join(' ') }}"
+    state: present
+    create: true
+    validate: 'sshd -T -f %s'
+    mode: '0644'
+  become: true
+  when: basic_security_ssh_allowed_users | length > 0
+  notify: basic_security_restart_sshd
+
+- name: Add configured groups allowed to connect over ssh
+  ansible.builtin.lineinfile:
+    dest: "{{ basic_security_ssh_config_path }}"
+    regexp: '^AllowGroups'
+    line: "AllowGroups {{ basic_security_ssh_allowed_groups | join(' ') }}"
+    state: present
+    create: true
+    validate: 'sshd -T -f %s'
+    mode: '0644'
+  become: true
+  when: basic_security_ssh_allowed_groups | length > 0
+  notify: basic_security_restart_sshd
+
+- name: Add configured user accounts to passwordless sudoers.
+  ansible.builtin.lineinfile:
+    dest: /etc/sudoers
+    regexp: '^{{ item }}'
+    line: '{{ item }} ALL=(ALL) NOPASSWD: ALL'
+    state: present
+    validate: 'visudo -cf %s'
+    mode: '0440'
+  become: true
+  with_items: "{{ basic_security_sudoers_passwordless }}"
+  when: basic_security_sudoers_passwordless | length > 0
+
+- name: Add configured user accounts to passworded sudoers.
+  ansible.builtin.lineinfile:
+    dest: /etc/sudoers
+    regexp: '^{{ item }}'
+    line: '{{ item }} ALL=(ALL) ALL'
+    state: present
+    validate: 'visudo -cf %s'
+    mode: '0440'
+  become: true
+  with_items: "{{ basic_security_sudoers_passworded }}"
+  when: basic_security_sudoers_passworded | length > 0
```

Comment:
I know you said you just copy-paste and droped the ubuntu part, but I would also argue that we don't need any of this. I would just remove it, together with the associated variables. It would be more lean and clean, and it reduces the possibility of someone accidentally adding the sudoers here.

Reply:
Point taken; I'll remove it.